src/vm.rs

```diff
@@ -309,6 +301,72 @@ impl<E: UserDefinedError, I: 'static + InstructionMeter> Executable<E, I> {
     }
 }
 
+/// Verified executable
```

Comment:
I really dislike this wrapper here because it is mostly an empty hull. Why don't we just add a "verified" boolean flag to the executable?

Reply:
I'd like to make to explicitly clear in the code whether we are working with a verified or unverified executable and a type difference makes that crystal.  Similar to message and sanitized message.

Reply:
So in other words you want it to be distinguished at compile time, thus the need for a type.
Are there any places where we want to skip the verification? Tests / benches maybe?
Because otherwise I would say: Just make verification mandatory.

Reply:
Well, I initially was going to internalize the verifier, but there are a lot of places in tests where we skip the verifier to perform deeper tests.  Also, in the loader, we split out the verifier so we can keep separate metrics for it vs the loader, etc...

Reply:
Here are the corresponding mon-repo changes: https://github.com/solana-labs/solana/pull/25593

Reply:
Part of the issue is that the verification is also specified in different places (assemble, from_elf, etc...).  This way it is done in only one way and must be done in order to create a vm.

Reply:
One more idea to get rid off the boilerplate methods here:

```Rust
#[derive(Debug, PartialEq)]
#[repr(transparent)]
pub struct VerifiedExecutable<E: UserDefinedError, I: InstructionMeter> {
    executable: Pin<Box<Executable<E, I>>>,
    _private_marker: (),
};

impl<E: UserDefinedError, I: InstructionMeter> VerifiedExecutable<E, I> {
    /// Verify an executable
    pub fn from_executable(
        executable: Pin<Box<Executable<E, I>>>,
        verifier: Verifier,
    ) -> Result<Self, EbpfError<E>> {
        verifier(executable.get_text_bytes().1, executable.get_config())?;
        Ok(Self { executable, _private_marker: () })
    }
}
```

So the inner `executable` can be publicly accessible (without accessor methods),
while the only way to get a `VerifiedExecutable` remains being its constructor.

Reply:
Does this mean that all users of `VerifiedExecutable` would have to things like this? `verified_executable.executable.get_config()`

What is your particular concern about the access functions?

Reply:
Yes, that would be the downside. My concern is that it duplicates the interfaces and they have to be kept in sync, potentially even their order (depending on other hacky things we do).

Reply:
Hmm, the compller will keep the dup interfaces in check...what other hacky things are you planning to do? ;-)

Reply:
On Discord there is a discussion about saving the compiled JIT executable to disk and currently it needs a reference to the executable, but I think that could be removed.